ToolBox/Benzinga/BenzingaDataConverter.cs

```diff
@@ -0,0 +1,346 @@
+
+                    if (string.IsNullOrWhiteSpace(mappedSymbol))
+                    {
+                        Log.Error($"BenzingaDataConverter.Process(): Failed to map old ticker {symbolTicker}. New ticker is null");
+                        continue;
+                    }
+
+                    instance.Symbols.Add(new BenzingaSymbolData()
+                    {
+                        Exchange = exchange,
+                        Sentiment = sentiment,
+                        Symbol = new Symbol(
+                            SecurityIdentifier.GenerateEquity(symbolTicker, Market.USA, mapSymbol: true, mapFileProvider: _mapFileProvider, mappingResolveDate: instance.PublicationDate),
+                            mappedSymbol
+                        )
+                    });
+                }
+
+                yield return instance;
+            }
+        }
+
+        /// <summary>
+        /// Writes the processed data into their respective files
+        /// </summary>
+        /// <param name="news">Enumerable containing news articles</param>
+        /// <param name="date">Date to convert for</param>
+        private void WriteToFile(IEnumerable<BenzingaNews> news, DateTime date)
+        {
+            var finalDirectory = new DirectoryInfo(Path.Combine(_destinationDirectory.FullName, "content", $"{date:yyyMMdd}"));
+            finalDirectory.Create();
+
+            foreach (var article in news)
+            {
+                var createdReference = false;
+
+                // For each ticker, add a reference to the article we're going to store under the "content" folder
+                foreach (var ticker in article.Symbols)
+                {
+                    // Invalid character in path
+                    if (ticker.Symbol.Value.Contains(":"))
+                    {
+                        Log.Trace($"BenzingaDataConverter.WriteToFile(): Ticker {ticker.Symbol.Value} contains invalid character ':'. Skipping");
+                        continue;
+                    }
+                    // ETFs have a special exchange tag associated with them.
+                    // Sometimes, we have an empty exchange for some tickers. Let's try and map them first before giving up on them.
+                    // Example: SPOT (Spotify) on 2018-10-01 has a missing exchange even though it has already IPO'd
+                    if (ticker.Exchange != "NASDAQ" && ticker.Exchange != "NYSE" && ticker.Exchange != "ETF" && !string.IsNullOrEmpty(ticker.Exchange))
+                    {
+                        Log.Trace($"BenzingaDataConverter.WriteToFile(): Ticker {ticker.Symbol.Value} is not in NYSE, NASDAQ, or is not an ETF. Skipping");
+                        continue;
+                    }
+                    // We can't write these tickers in Windows, so we'll skip them
+                    if (_isWindows && _forbiddenTickers.Contains(ticker.Symbol.Value))
```

Comment:
totally style and feel free to ignore this, but my preference would be to use `OS.IsWindows` directly

Reply:
Just a note for you 😄 : I avoided using `OS.IsWindows` since there is close to absolutely no possibility that we will have the OS switch on us while the program is running. In addition to that, since the property would perform a [few allocations per call](https://referencesource.microsoft.com/#mscorlib/system/environment.cs,7d2f1469d916fc63) (albeit probably in a low heap generation), it wouldn't matter. But what I'm concerned about is possible syscalls being made trying to identify the platform causing the processing to slow down since we're leaving userspace, especially in a hotter path.

Agreed with you that OS.IsWindows is much cleaner, but those were the design considerations I made when I chose to cache the `OS.IsWindows` call.